src/server/pfs/server/driver.go

```diff
@@ -558,35 +568,33 @@ func (d *driver) makeCommit(pachClient *client.APIClient, ID string, parent *pfs
 					return err
 				}
 			}
-		} else {
-			isStartCommit = true
-			if err := d.openCommits.ReadWrite(stm).Put(newCommit.ID, newCommit); err != nil {
-				return err
-			}
-		}
 
-		if treeRef != nil {
+			// now 'treeRef' is guaranteed to be set
 			newCommitInfo.Tree = treeRef
 			newCommitInfo.SizeBytes = uint64(tree.FSSize())
 			newCommitInfo.Finished = now()
-		}
 
-		// Update the repo size if we're on the master branch and this is not
-		// an open commit
-		if !isStartCommit && branch == "master" {
-			repoInfo.SizeBytes = newCommitInfo.SizeBytes
+			// If we're updating the master branch, also set the repo size (persisted
+			// below, along with new branch)
+			if branch == "master" {
+				repoInfo.SizeBytes = newCommitInfo.SizeBytes
+			}
+		} else {
+			if err := d.openCommits.ReadWrite(stm).Put(newCommit.ID, newCommit); err != nil {
+				return err
+			}
```

Comment:
Likewise with the above comment; this used to only hit if records was nil

Reply:
I think it should still work that way. The indentation is getting hard to read in GitHub (at least for me), but from my editor, this section is basically:
```
		// 1. Write 'newCommit' to 'openCommits' collection OR
		// 2. Finish 'newCommit' (if treeRef != nil or records != nil); see
		//    "FinishCommit case" above)
		if treeRef != nil || records != nil {
			if records != nil {
				...
			}

			// If we're updating the master branch, also update the repo size (see
			// "Update repoInfo" below)
			if branch == "master" {
				repoInfo.SizeBytes = newCommitInfo.SizeBytes
			}
		} else {
			if err := d.openCommits.ReadWrite(stm).Put(newCommit.ID, newCommit); err != nil {
				return err
			}
		}
```